src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionListener.cs

```diff
@@ -50,26 +50,30 @@ internal sealed class SocketConnectionListener : IConnectionListener
                 for (var i = 0; i < _settingsCount; i++)
                 {
                     var transportScheduler = options.UnsafePreferInlineScheduling ? PipeScheduler.Inline : new IOQueue();
+                    var awaiterScheduler = OperatingSystem.IsWindows() ? transportScheduler : PipeScheduler.Inline;
 
                     _settings[i] = new Settings
                     {
                         Scheduler = transportScheduler,
                         InputOptions = new PipeOptions(_memoryPool, applicationScheduler, transportScheduler, maxReadBufferSize, maxReadBufferSize / 2, useSynchronizationContext: false),
-                        OutputOptions = new PipeOptions(_memoryPool, transportScheduler, applicationScheduler, maxWriteBufferSize, maxWriteBufferSize / 2, useSynchronizationContext: false)
+                        OutputOptions = new PipeOptions(_memoryPool, transportScheduler, applicationScheduler, maxWriteBufferSize, maxWriteBufferSize / 2, useSynchronizationContext: false),
+                        SocketSenderPool = new SocketSenderPool(awaiterScheduler)
```

Comment:
Did you try a single pool for all IOQueues?

Reply:
Nope.

Reply:
The problem is its tied to the scheduler which is tied to the IOQueue

Reply:
You could make `SocketAwaitableEventArgs._ioScheduler` resettable and make the `PipeScheduler` an argument to `Send/ReceiveAsync` much like `Socket` if you really wanted, but I think having a pool per IOQueue makes sense to reduce contention.